### PR TITLE
[SNMP] Minimal SNMP traps component

### PR DESCRIPTION
**Pull request by dplepage-dd**
### What does this PR do?

This does the bare minimum work necessary to make the snmp traps server run in a component. It updates the snmp traps server so that it takes the configuration and logger as arguments instead of assuming globals, and creates a tiny component that just invokes the global `traps.StartServer` and `traps.StopServer`.

### Motivation
NDMII-287 - we want the server to be managed by the component system. Originally we wrote a total componentization in #19452, but we decided that this was too many changes at once, so this is the smallest PR that just gets the traps server to be managed by the component lifecycle.

### Describe how to test/QA your changes

See internal instructions for QAing snmp traps. Existing QA should be fine, as the behavior of the traps listener is not expected to change.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/snmp/traps/config.go [hunk 1]
```diff
@@ -104,13 +105,17 @@ func (c *Config) Addr() string {
 }
 
 // BuildSNMPParams returns a valid GoSNMP params structure from configuration.
-func (c *Config) BuildSNMPParams() (*gosnmp.GoSNMP, error) {
+func (c *Config) BuildSNMPParams(logger log.Component) (*gosnmp.GoSNMP, error) {
+	var snmpLogger gosnmp.Logger
+	if logger != nil {
+		snmpLogger = gosnmp.NewLogger(&trapLogger{logger: logger})
+	}
```
Comment (AlexandreYang) [inline pkg/snmp/traps/config.go:112]: **❓ question:** ‏Is it safe to continue the program if `snmpLogger` is nil ?

Comment (AlexandreYang) [reply pkg/snmp/traps/config.go:112]: Is it expected for `logger log.Component` to be nil? Shall we stop the program when logger is nil?

**Review by AlexandreYang — COMMENTED** [review]
LGTM overall, only few minor comments :)

File: pkg/snmp/traps/config.go (hunk 1 above)
Comment (dplepage-dd) [reply pkg/snmp/traps/config.go]: It shouldn't be nil. I've changed this to just create the logger, which matches the existing behavior.

**Comment by dplepage-dd** [comment]
Closing this in favor of #19789.

**Pull request closed without merging**